core/src/main/scala/org/apache/spark/util/ClosureCleaner.scala

```diff
@@ -205,8 +222,7 @@ private[spark] object ClosureCleaner extends Logging {
 
     // A list of enclosing objects and their respective classes, from innermost to outermost
     // An outer object at a given index is of type outer class at the same index
-    val outerClasses = getOuterClasses(func)
-    val outerObjects = getOuterObjects(func)
```

Comment:
You should remove these methods (or, if they're useful in their own right, re-implement them in terms of the new method). Now there's three implementations of the same logic.